core/src/test/scala/unit/kafka/server/MetadataCacheTest.scala

```diff
@@ -132,7 +165,8 @@ class MetadataCacheTest {
       partitionStates.asJava, brokers.asJava).build()
     cache.updateCache(15, updateMetadataRequest)
 
-    val topicMetadatas = cache.getTopicMetadata(Set(topic), listenerName)
+    val errorUnavailableListeners = expectedError == Errors.LISTENER_NOT_FOUND_ON_LEADER
```

Comment:
Would it be clearer if this was passed as a boolean? We could possibly want to test this method with the boolean true and `LEADER_NOT_AVAILABLE` because the listener was present even, the leader was not and we were using the API with the boolean == true. Is that a testing gap actually?